### PR TITLE
Update the readme

### LINKED ISSUE #745: Add documentation for connection/user count
The new connection count feature is not documented well enough. At least the following information should be documented somewhere:
- What is the meaning of the log keywords, e.g. `current_connection_total`?
- What is the interface of the http route? Where is it reachable/what is returned in what format?

Without this information, it will be hard to implement this feature in the client.

### PR DESCRIPTION
**Pull request by ostcar**
Fixes #745

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,38 +24,32 @@ It also needs a running postgres and redis instance. You can start one with:
...
 
 ### With Docker
 
-The docker build uses the auth token as default. Either configure it to use the
-auth-fake services (see environment variables below) or make sure the service
-inside the docker container can connect to the auth service. For example with
-the docker argument --network host. The auth-secrets have to given as a file.
+Make sure the service inside the docker container can connect to the auth
+service, postgres and redis. For example with the docker argument --network
+host.
```
Comment (jsangmeister) [inline README.md:46]:
```suggestion
service, postgres and redis, for example with the docker argument `--network
host`.
```

File: README.md [hunk 2]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
 It also supports the attributes `single=1` and the normal autoupdate body.
 
 
+### Connection Count
+
+The autoupdate services saves, how many connections each user currently open.
```
Comment (jsangmeister) [inline README.md:200]:
```suggestion
The autoupdate services saves how many connections are currently open to each user.
```

File: README.md [hunk 3]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+The save-interval can be defined with the environment variable
```
Comment (jsangmeister) [inline README.md:201]:
```suggestion
The save interval can be defined with the environment variable
```

File: README.md [hunk 4]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+METRIC_SAVE_INTERVAL. The default is 5 minutes.
```
Comment (jsangmeister) [inline README.md:202]:
```suggestion
`METRIC_SAVE_INTERVAL`. The default is 5 minutes.
```

File: README.md [hunk 5]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+
+The values are saved for each instance of the autoupdate-service. So it is
```
Comment (jsangmeister) [inline README.md:204]:
```suggestion
The values are saved for each instance of the autoupdate service. So it is
```

File: README.md [hunk 6]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+possible to access all open connection for every autoupdate-service-instance in
```
Comment (jsangmeister) [inline README.md:205]:
```suggestion
possible to access all open connection for every instance of the autoupdate service in
```

File: README.md [hunk 7]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+the same cloud.
+
+`curl "localhost:9012/service/autoupdate/connection_count`
```
Comment (jsangmeister) [inline README.md:208]:
```suggestion
`curl "localhost:9012/service/autoupdate/connection_count"`
```

File: README.md [hunk 8]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+
+It returns a json-dictonary like this:
```
Comment (jsangmeister) [inline README.md:210]:
```suggestion
It returns a JSON dictonary like this:
```

File: README.md [hunk 9]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+
+`{"0":15,"1":4,"2":3}`
+
+The key is a user ID and the value amount of currently open connections. User-ID
```
Comment (jsangmeister) [inline README.md:214]:
```suggestion
The key is a user ID and the value is the amount of currently open connections. User ID
```

File: README.md [hunk 10]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+`0` is the anonymous user. It the example above, the anonymous user has 15 open
+connections, the user with the id 1 has 4 open connections and the user with the
```
Comment (jsangmeister) [inline README.md:216]:
```suggestion
connections, the user with the ID 1 has 4 open connections and the user with the
```

File: README.md [hunk 11]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+ID 2 has 3 open connection.
+
+Users can only access this page, if they have the organization management level
```
Comment (jsangmeister) [inline README.md:219]:
```suggestion
Users can only access this page if they have the organization management level
```
Which organization management level? I guess `can_manage_organization`?

File: README.md [hunk 12]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+or higher.
+
+
+## Metric
+
+The autoupdate-service logs some metric values. The interval can be set with the
```
Comment (jsangmeister) [inline README.md:225]:
```suggestion
The autoupdate service logs some metric values. The interval can be set with the
```

File: README.md [hunk 13]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+environment varialbe METRIC_INTERVAL.
```
Comment (jsangmeister) [inline README.md:226]:
```suggestion
environment variable `METRIC_INTERVAL`.
```

File: README.md [hunk 14]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+    "datastore_cache_key_len": 343,
+    "datastore_cache_size": 3114,
+    "runtime_goroutines": 42
+}
+```
+
+The values are:
+
+* `connected_users_anonymous_connections`: Number of connections from the
+  anonymous users from all autoupdate-instances.
```
Comment (jsangmeister) [inline README.md:249]:
```suggestion
  anonymous users from all autoupdate instances.
```

File: README.md [hunk 15]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+* `connected_users_average_connections`: Average connection count for each user
+  except for anonymous user.
+* `connected_users_current`: Amount of connected users that have at least one
+  open connection.
+* `connected_users_current_local`: Amount of connected users that have at least
+  one open connection of this instance.
+* `connected_users_total`: Amount of different users that are currently
+  connected or where connection since the start of the autoupdate-service.
```
Comment (jsangmeister) [inline README.md:257]:
```suggestion
  connected or were connected since the autoupdate service was started.
```

File: README.md [hunk 16]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+* `connected_users_total_local`: Same as connected_users_total but only for this
```
Comment (jsangmeister) [inline README.md:258]:
```suggestion
* `connected_users_total_local`: Same as `connected_users_total`, but only for this
```

File: README.md [hunk 17]
```diff
@@ -201,6 +195,75 @@ The autoupdate service provides an internal route to return fields for a defined
...
+instance. one open connection of this instance.
```
Comment (jsangmeister) [inline README.md:259]:
```suggestion
instance.
```

**Review by jsangmeister — CHANGES_REQUESTED** [review]
Sorry for the nitpicking :)

**Comment by ostcar** [comment]
@jsangmeister could you open a new PR that I can merge instead of this?

**Comment by jsangmeister** [comment]
@ostcar I just applied all the changes here. From my side, it can be merged now, if you have no further comments.

**Pull request merged**